src/utils/warehouse/notion/kpi_formatter.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+require 'bas/utils/notion/request'
+
+module Utils
+  module Warehouse
+    module Notion
+      module Formatter
+        ##
+        # This class formats Notion KPI records into a standardized hash format.
+        # It also fetches nested stats from a child database within each KPI page.
+        class KpiFormatter < Base
+          def format
+            {
+              external_kpi_id: extract_id,
+              description: extract_rich_text('Description'),
+              status: extract_select('Status'),
+              current_value: extract_number('Current Value'),
+              percentage: extract_formula_number('Percentage'),
+              target_value: extract_number('Target Value'),
+              external_domain_id: extract_relation('Domain')
+            }
+          end
+
+          class << self
+            ##
+            # Fetches and formats KPIs from Notion, including their stats.
+            # It retrieves the stats from a child database linked to each KPI page.
+            def fetch_stats_for_kpis(records, secret:, filter_body: {})
+              records.map do |kpi_record|
+                formatted_kpi = new(kpi_record).format
+
+                kpi_page_id = kpi_record['id']
```

Comment:
Since the kpi_page_id is only used as a parameter in the find_stats_database_id function, you could pass the value directly, which reduces the number of lines.
```ruby
stats_db_id = find_stats_database_id(kpi_record['id'], secret: secret)
```